ably/hosts.go

```diff
@@ -0,0 +1,233 @@
+package ably
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ably/ably-go/ably/internal/ablyutil"
+)
+
+// RSC15
+type restHosts struct {
+	primaryFallbackHost string
+	opts                *clientOptions
+	cache               *hostCache
+	visitedHosts        []string
```

Comment:
It's customary to use `map[T]struct{}` or `map[T]bool` as a `Set<T>` type. It also makes `ablyutil.Contains` unnecessary.

Reply:
Updated with map based set type